src/debugger/gdb-stub.c

```diff
@@ -29,6 +29,30 @@ enum {
 	MACH_O_ARM_V4T = 5
 };
 
+static const char* TARGET_XML = "<target version=\"1.0\">"
+                                "<architecture>armv4t</architecture>"
+                                "<osabi>none</osabi>"
+                                "<feature name=\"org.gnu.gdb.arm.core\">"
+                                "<reg name=\"r0\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r1\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r2\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r3\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r4\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r5\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r6\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r7\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r8\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r9\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r10\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r11\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"r12\" bitsize=\"32\" type=\"uint32\"/>"
+                                "<reg name=\"sp\" bitsize=\"32\" type=\"data_ptr\"/>"
+                                "<reg name=\"lr\" bitsize=\"32\"/>"
```

Comment:
This is sometimes used as a generic int, but sometimes not. Is there a strong reason for it to be labeled either way, or not at all?

Reply:
These are just to tell the width of the registers. GDB's definition have the same things, but defining them explicitly allows for example to have GDB not try to check for the FPR because they aren't present.

So really, it's because that's how GDB does it vanilla.